tests/builtins/functions/test_update_state.py

```diff
@@ -44,10 +51,58 @@ def test_empty_event():
         run_function("update-state", {})
 
 
-def test_success(event, statedb):
+@pytest.fixture
+def workflow_event_topic(sns):
+    return sns.create_topic(Name="app-cirrus-workflow-event")["TopicArn"]
+
+
+@pytest.mark.parametrize(
+    "wf_event_enabled,sfn_state", product((True, False), SfnStatus._member_names_)
+)
+def test_workflow_event_notification(
+    event, statedb, workflow_event_topic, wf_event_enabled, sfn_state, monkeypatch
+):
+    """This tests that workflow events are properly published (if `wf_event_enabled`),
+    with the associated StepFunctions State.  It also verifies that the record is added
+    to the `statedb`.  The content of that record is validated in separate
+    `test_{StateEnum}` tests.
+
+    N.B. the tests end with `wf_event_enabled=False`, such that the loaded
+    lambda_function module does not have the SNS publishing enabled, and the other tests
+    do not require the `workflow_event_topic` fixture to create the topic (as it will
+    not be used).
+    """
+    if wf_event_enabled:
+        expected_msg_count = 1
+        monkeypatch.setenv("CIRRUS_WORKFLOW_EVENT_TOPIC_ARN", workflow_event_topic)
+    else:
+        expected_msg_count = 0
+        monkeypatch.delenv("CIRRUS_WORKFLOW_EVENT_TOPIC_ARN", raising=False)
+    if "cirrus.builtins.functions.update-state.lambda_function" in sys.modules:
+        importlib.reload(
+            sys.modules["cirrus.builtins.functions.update-state.lambda_function"]
+        )
```

Comment:
This is to deal with the state of the `wf_event_manager.event_publisher` which is initialized at import time.

Reply:
i hadn't seen this pattern before, good to know

Reply:
unresolving because i commented on a note for other reviewers

Reply:
The reload has been made unnecessary by changes in #263